samcli/commands/local/lib/local_lambda.py

```diff
@@ -141,6 +142,12 @@ def _make_env_vars(self, function):
         #
         # Standard format is {FunctionName: {key:value}, FunctionName: {key:value}}
         # CloudFormation parameter file is {"Parameters": {key:value}}
+
+        for fn in self.env_vars_values:
+            if not isinstance(self.env_vars_values.get(fn), dict):
+                LOG.debug("Environment variables overrides data is not in a recognized format")
```

Comment:
Add context on what failed here. Customers can then use `--debug` to help them in understanding the issue. This will also be useful for us when issues come in with `--debug` data to understand where the problem is.

Reply:
@sthulb Can you please address this?